File: Utilities/Matlab/scripts/fan_curve.m
% McDermott and Floyd
% 10-3-12
% fan_curve.m
%
% Makes figure for HVAC Fan Parameters section of the FDS Users Guide

close all
clear all

plot_style
figure
set(gca,'Units',Plot_Units)
set(gca,'Position',[Plot_X Plot_Y Plot_Width Plot_Height])

vdot_max = 10;
dp_max = 500;

dp = -1000:1000;

vdot = vdot_max*sign(dp_max-dp).*sqrt(abs(dp-dp_max)/dp_max);
vdot1 = 10;
XLegendStr{1}='constant volume';
XLegendStr{2}='quadratic';
XLegendStr{3}='user fan curve';
K(1)=plot(vdot1*ones(1,length(dp)),dp,'r-','LineWidth',2); hold on
K(2)=plot(vdot,dp,'k-','LineWidth',2);
i=0;
for dp=-1000:200:1000
    i=i+1;
    rampx(i) = vdot_max*sign(dp_max-dp).*sqrt(abs(dp-dp_max)/dp_max);
    rampy(i) = dp;
end
K(3)=plot(rampx,rampy,'b-','LineWidth',2);
set(gca,'FontName',Font_Name)
set(gca,'FontSize',Title_Font_Size)
set(gca,'XGrid','on')
set(gca,'YGrid','on')
axis([-10 20 -1000 1000])
xlabel('Volume Flow Rate (m^3/s)','Interpreter',Font_Interpreter,'FontName',Font_Name,'FontSize',Label_Font_Size)
ylabel('Static Pressure (Pa)','Interpreter',Font_Interpreter,'FontName',Font_Name,'FontSize',Label_Font_Size)
lh=legend(K,XLegendStr,'Location','Southwest');
set(lh,'FontSize',Key_Font_Size)
% print to pdf
plot_dir = '../../Manuals/FDS_User_Guide/SCRIPT_FIGURES/';
set(gcf,'Visible',Figure_Visibility);
set(gcf,'Units',Paper_Units);
set(gcf,'PaperUnits',Paper_Units);
set(gcf,'PaperSize',[Paper_Width Paper_Height]);
set(gcf,'Position',[0 0 Paper_Width Paper_Height]);
print(gcf,'-dpdf',[plot_dir,'fan_curve'])

File: Utilities/Python/scripts/fan_curve.py

# Makes figure for HVAC Fan Parameters section of the FDS Users Guide

import numpy as np
import matplotlib.pyplot as plt
import fdsplotlib

plot_style = fdsplotlib.get_plot_style('fds')

pltdir = '../../Manuals/FDS_User_Guide/SCRIPT_FIGURES/'

vdot_max = 10
dp_max = 500
dp = np.arange(-1000, 1001, 1)

# Constant volume flow rate
vdot1 = 10

# Plot quadratic fan curve (black line)
vdot = vdot_max * np.sign(dp_max - dp) * np.sqrt(np.abs(dp - dp_max) / dp_max)

# Create user fan curve (ramp) data points
rampx = []
rampy = []
i = 0
for dp_val in range(-1000, 1001, 200):
    rampx.append(vdot_max * np.sign(dp_max - dp_val) * np.sqrt(np.abs(dp_val - dp_max) / dp_max))
    rampy.append(dp_val)
    i = i + 1

fig = fdsplotlib.plot_to_fig(x_data=[vdot1,vdot1], y_data=[-1000,1000], marker_style='r-', data_label='constant volume',
                             x_min=-10, x_max=20, y_min=-1000, y_max=1000,
                             x_label='Volume Flow Rate (m$^3$/s)',
                             y_label='Static Pressure (Pa)')

fdsplotlib.plot_to_fig(x_data=vdot, y_data=dp, marker_style='k-', data_label='quadratic', figure_handle=fig)
fdsplotlib.plot_to_fig(x_data=rampx, y_data=rampy, marker_style='b-', data_label='user fan curve', figure_handle=fig)

# Enable grid
ax = plt.gca()
ax.grid(True, which='both', axis='both')

plt.savefig(pltdir + 'fan_curve.pdf', format='pdf')
plt.close()

